### ISSUE TITLE
Java enums with java annotations - error producing SchemaFor

### ISSUE THREAD

**Issue by klass-ivan** [sksamuel/avro4s#302]
```java
package org.test

@org.apache.avro.specific.AvroGenerated
public enum MyType {
 A, B, C  ;
 /* ... */ 
}
```
`AvroSchema[MyType]`

```   
expected start of definition
        at com.sksamuel.avro4s.SchemaFor$$anon$13.$anonfun$schema$1(SchemaFor.scala:170)
        at com.sksamuel.avro4s.SchemaFor$$anon$13.schema(SchemaFor.scala:167)
        at com.sksamuel.avro4s.SchemaFor$.$anonfun$buildField$3(SchemaFor.scala:373)
        at com.sksamuel.avro4s.SchemaFor$.buildField(SchemaFor.scala:373)
        at com.sksamuel.avro4s.SchemaFor$.schemaFieldNoDefault(SchemaFor.scala:313)
```

which points to:
```scala
  implicit def javaEnumSchemaFor[E <: Enum[_]](implicit tag: ClassTag[E]): SchemaFor[E] = new SchemaFor[E] {
    override def schema: Schema = {

      val annos = tag.runtimeClass.getAnnotations.toList.map { a =>
        val tb = currentMirror.mkToolBox()
         // the following line is failing
        val args = tb.compile(tb.parse(a.toString)).apply() match { ...
```

the `a.toString` is `@org.apache.avro.specific.AvroGenerated()` and `tb.parse(a.toString)` fails:
```
reflective compilation has failed:

expected start of definition
```

The suggestion is to skip java annotations, may be?

**Comment by sksamuel** [comment]
Yes I think java annotations could be skipped.

**Comment by sksamuel** [comment]
Can you try the 3.0.x snapshot from 
https://oss.sonatype.org/content/repositories/snapshots/com/sksamuel/avro4s/avro4s-core_2.12/
This should be fixed and is available for 2.11 and 2.22

**Comment by sksamuel** [comment]
Closing this as it works with 3.0.x
